### PR TITLE
Create Account

### PR DESCRIPTION
**Pull request by cxu2**
- Fixed proposed changes from last PR
- Created Account runnable and tested
- Added identity.sqlite as testing db
- Added .travis.yml
- Set-cookie not included
- Updated account partially written

### CODE REVIEW

**Comment by adavis10006** [comment]
Can you remove the comment codes, other LGTM.

**Comment by IanChen83** [comment]
Instead of storing SQLite database in the source control, I think it would be better to write a script to create and initialize the database to the desired state.

**Comment by DandilionLau** [comment]
> Can you remove the comment codes? Others are LGTM.

Since the updateAccount function in this PR has not been fully developed, all comments will be removed when updateAccount is ready (1-2 PR later estimated).

**Comment by DandilionLau** [comment]
> Instead of storing SQLite database in the source control, I think it would be better to write a script to create and initialize the database to the desired state.

Will remove in next PR and add a bash script to create DB upon testing.

File: src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java [hunk 1]
```diff
@@ -2,20 +2,94 @@
...
-    public SqlAccountUpdate(@Nonnull SqlSessionFactory sqlSessionFactory) {
+    public SqlAccountUpdate(@Nonnull SqlSessionFactory sqlSessionFactory, @Nonnull String id) {
         this.sqlSessionFactory = sqlSessionFactory;
+
+        try(SqlSession session = sqlSessionFactory.openSession()){
+            AccountMapper mapper = session.getMapper(AccountMapper.class);
+            this.account = mapper.getAccount(id);
+        }catch (Exception e) {
+            this.account = new AccountModel();
+        }
```
Comment (sleepingpig) [inline src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java]: Why we need to get the account from db first before updating?

**Comment by sleepingpig** [comment]
Please remove the sqlite db file in this PR. Not in next one.

File: src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java (hunk 1 above)
Comment (DandilionLau) [reply src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java]: Sorry I didn't discuss the implementation details with you. 

Idea: get the old email, password, description -> replace some of them with new ones -> write in a unified SQL operation: `UPDATE account SET email=email, password=password, description=description.`

Do you mean that we can assume the request sent by frontend must contain all three editable fields? If yes, we can drop the getAccount() method.

**Comment by DandilionLau** [comment]
> Please remove the sqlite db file in this PR. Not in next one.

Apology. Will fix everything tomorrow morning in this PR before merge. 
(I previously thought PR was not editable once proposed, now I figure it out :ok_hand: )

File: src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java (hunk 1 above)
Comment (sleepingpig) [reply src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java]: Not all fields would be available every time.
To ignore null properties using MyBatis, see http://www.mybatis.org/mybatis-3/dynamic-sql.html

Comment (DandilionLau) [reply src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java]: I see. Thanks! 👍

**Comment by DandilionLau** [comment]
Issues have been fixed today:
⋅⋅* DB file has been removed and each time travisCI will run .scripts/buidDB.sh to create a new one
⋅⋅* comments have been removed
⋅⋅* getAccount in updateAccount has been removed

**Pull request merged**